packages/frontend-shared/src/gql-components/promo/Promo.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div class="promo-box flex flex-col items-center border-gray-100 rounded-lg w-full max-w-[1004px] m-auto">
+    <slot name="header" />
+    <Slideshow class="w-full">
+      <template #default="{ step, goBack, goForward, reset }">
+        <slot
+          name="cards"
+          :step="step"
+          :go-back="goBack"
+          :go-forward="goForward"
+          :reset="reset"
+        />
+      </template>
+    </Slideshow>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import Slideshow from '../../components/Slideshow.vue'
+import { watch } from 'vue'
+import { gql, useMutation, useQuery } from '@urql/vue'
+import { nanoid } from 'nanoid'
+
+import { PromoDocument, Promo_PromoSeenDocument } from '../../generated/graphql'
+
+gql`
+fragment _Promo on Query {
+  currentProject {
+    id
+    savedState
+  }
+}
+`
+
+gql`
+query Promo {
+  ..._Promo
+}
+`
+
+gql`
+mutation Promo_PromoSeen($campaign: String!, $messageId: String!, $medium: String!, $cohort: String) {
+  recordEvent(campaign: $campaign, messageId: $messageId, medium: $medium, cohort: $cohort)
+}
+`
+
+const props = defineProps<{
+  campaign: string
+  medium: string
+  cohort?: string
+}>()
+
+const query = useQuery({ query: PromoDocument })
+const promoSeenMutation = useMutation(Promo_PromoSeenDocument)
+
+const promoInstanceId = nanoid()
+
+watch([query.data.value], ([queryResult]) => {
```

Comment:
Seems like just want to run this once - as soon as the query data is available - is that correct?

Could we do

```ts
useQuery().then(res => {
  // we know it's resolved, do the check and mutation
})
```

Eg - do we expect this re-trigger (do we need to `watch`)?

Reply:
Much better, didn't realize `useQuery` was Promise-like 👍